pkg/datastore/datastore.go

```diff
@@ -43,18 +42,57 @@ type RevisionChanges struct {
 	Changes  []*core.RelationTupleUpdate
 }
 
+type Reader interface {
+	// QueryTuples reads relationships starting from the resource side.
+	QueryRelationships(
+		ctx context.Context,
+		filter *v1.RelationshipFilter,
+		options ...options.QueryOptionsOption,
+	) (RelationshipIterator, error)
+
+	// ReverseQueryRelationships reads relationships starting from the subject.
+	ReverseQueryRelationships(
+		ctx context.Context,
+		subjectFilter *v1.SubjectFilter,
+		options ...options.ReverseQueryOptionsOption,
+	) (RelationshipIterator, error)
+
+	// ReadNamespace reads a namespace definition and version and returns it, and the revision at
```

Comment:
ReadNamespace reads a namespace definition, its version, and its last updated revision